DataBiosphere/leonardo
# [IA-4057] Check that user is enabled when showing empty lists

…akes no resource permission checks (because resources returned are an empty list)

When an endpoint would return an empty list, and the user is not enabled in Sam, fail with 401 Forbidden, don't serve an empty list.

Validated with trivial unit test for SamAuthProvider + testing in my BEE with a user which has not accepted ToS.

Steps to validate:
- Create resources (disks, apps, runtimes) in workspaces (Google and Azure) using an authenticated user.
- With an unauthenticated user (you've signed up, but have not accepted terms of service):
	- Authorize through Leonardo Swagger (e.g. https://leonardo.mnolting.bee.envs-terra.bio)
	- Make a request to an endpoint (the issue was seen on endpoints without a resource context (e.g. https://leonardo.mnolting.bee.envs-terra.bio/#/apps/listApp) which returned no resources (e.g. when filtered by _labels nonexistent=bar))
	- See that requests matching any resources are rejected with a 403 (existing behavior)
	- See that requests matching no resources are rejected with a 401 (bugfix as of this PR)

Note also the ProxyService issue I outlined in comments & https://broadworkbench.atlassian.net/browse/IA-4131.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:


In all cases:
